### PR TITLE
feat(ci_builder): Adding nvm

### PR DESCRIPTION
**Pull request by blmalone**

### CODE REVIEW

**Comment by tynes** [comment]
We are moving away from having JS in the monorepo. Why are you adding `nvm`?

File: ops/docker/ci-builder/Dockerfile [hunk 1]
```diff
@@ -157,3 +157,15 @@ COPY --from=rust-build /root/.rustup /root/.rustup
 # copy the rust installation, alongside the installed toolchains
 COPY --from=rust-build /root/.cargo/bin /root/.cargo/bin
 COPY --from=rust-build /root/.rustup /root/.rustup
+
```
Comment (blmalone) [inline ops/docker/ci-builder/Dockerfile]: Don't change this Dockerfile

Comment (blmalone) [reply ops/docker/ci-builder/Dockerfile]: 1. Create new folder with new dockerfile
2. Create a Dockerfile that's mainly empty but start from the ci-builder at a given version.
3. Then add nvm in the new Dockerfile

File: docker-bake.hcl [hunk 2]
```diff
@@ -232,6 +232,14 @@ target "ci-builder-rust" {
   tags = [for tag in split(",", IMAGE_TAGS) : "${REGISTRY}/${REPOSITORY}/ci-builder-rust:${tag}"]
 }
 
+target "ci-builder-security-tools" {
+  dockerfile = "./ops/docker/ci-builder/Dockerfile"
```
Comment (blmalone) [inline docker-bake.hcl]: Reference new dockerfile that's been created.

File: ops/tag-service/tag-tool.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 SERVICES  = [
     'ci-builder',
     'ci-builder-rust',
+    'ci-builder-security-tools',
```
Comment (blmalone) [inline ops/tag-service/tag-tool.py:8]: Add a starting version for this too

File: ops/docker/ci-builder/Dockerfile (hunk 1 above)
Comment (blmalone) [reply ops/docker/ci-builder/Dockerfile]: Make sure that I pass the platform when I am building locally. 

`--platform linux/amd64`

Comment (blmalone) [reply ops/docker/ci-builder/Dockerfile]:
```
docker buildx bake \                                                                              
    --progress plain \
    --platform linux/amd64 \
    --load \
    -f docker-bake.hcl \
    ci-builder-security-tools
```

Comment (blmalone) [reply ops/docker/ci-builder/Dockerfile]: docker buildx bake \
    --progress plain \
    --load \
    -f docker-bake.hcl \
    --platform linux/amd64 \
    ci-builder-security-tools

Comment (blmalone) [reply ops/docker/ci-builder/Dockerfile]: `docker buildx bake --progress plain --load -f docker-bake.hcl --platform linux/amd64 ci-builder-security-tools`

**Comment by blmalone** [comment]
@tynes 

> We are moving away from having JS in the monorepo. Why are you adding nvm?

We are making use of the ci-builder in https://github.com/ethereum-optimism/security-tools. 

More specifically, to be able to build older commits of the op-contracts (as part of the _Block History Integrity Checks_ project), we need to switch node versions. I'm adding this to the ci-builder so that we can use that in our CI.

**Comment by blmalone** [comment]
Closing this PR. Opting to install `nvm` in CI directly instead of creating a new security base image. If in the future we find ourselves needing to manually install many deps as part of CI then we can revisit this functionality.

**Pull request closed without merging**